quartz/plugins/emitters/contentIndex.ts

```diff
@@ -116,50 +121,67 @@ export const ContentIndex: QuartzEmitterPlugin<Partial<Options>> = (opts) => {
-            content: file.data.text ?? "",
-            richContent: opts?.rssFullHtml
-              ? escapeHTML(toHtml(tree as Root, { allowDangerousHtml: true }))
-              : undefined,
-            date: date,
-            description: file.data.description ?? "",
-          })
+      const feedIndices: Map<String, ContentIndex> = new Map()
+
+      // bfahrenfort: ts can't see the expansion of opts above that guarantees a non-null feedDirectories
+      const directories =
+        opts?.feedDirectories == null ? defaultOptions.feedDirectories : opts.feedDirectories
+
+      for (const feed of directories) {
+        const linkIndex: ContentIndex = new Map()
+        for (const [tree, file] of content) {
+          const slug = file.data.slug!
+
+          const date = getDate(ctx.cfg.configuration, file.data) ?? new Date()
+          if (
+            (opts?.includeEmptyFiles || (file.data.text && file.data.text !== "")) &&
+            (slug.startsWith(feed) || feed == "index")
+          ) {
+            linkIndex.set(slug, {
+              title: file.data.frontmatter?.title!,
+              links: file.data.links ?? [],
+              tags: file.data.frontmatter?.tags ?? [],
+              content: file.data.text ?? "",
+              richContent: opts?.rssFullHtml
+                ? escapeHTML(toHtml(tree as Root, { allowDangerousHtml: true }))
+                : undefined,
+              date: date,
+              description: file.data.description ?? "",
+            })
+          }
         }
+        feedIndices.set(feed, linkIndex)
       }
 
       if (opts?.enableSiteMap) {
         emitted.push(
           await write({
             ctx,
-            content: generateSiteMap(cfg, linkIndex),
+            // bfahrenfort: "index" is guaranteed non-null
+            // see directories instantiation and feedIndices.set iterating over directories
+            content: generateSiteMap(cfg, feedIndices.get("index")!),
             slug: "sitemap" as FullSlug,
             ext: ".xml",
           }),
         )
       }
 
       if (opts?.enableRSS) {
-        emitted.push(
-          await write({
+        directories.map(async (feed) => {
+          const emittedFeed = await write({
             ctx,
-            content: generateRSSFeed(cfg, linkIndex, opts.rssLimit),
-            slug: "index" as FullSlug,
+            // bfahrenfort: we just generated a feedIndices entry for every directories entry, guaranteed non-null
+            content: generateRSSFeed(cfg, feedIndices.get(feed)!, opts?.rssLimit),
+            slug: feed as FullSlug,
             ext: ".xml",
-          }),
-        )
+          })
+          emitted.push(emittedFeed)
+        })
       }
 
       const fp = joinSegments("static", "contentIndex") as FullSlug
       const simplifiedIndex = Object.fromEntries(
-        Array.from(linkIndex).map(([slug, content]) => {
+        Array.from(feedIndices.get("index") ?? []).map(([slug, content]) => {
```

Comment:
same as above, this seems super brittle. I think we can just get the first index above then do the logic check there.

Reply:
Didn't notice the pattern, I'll refactor this and the other reference

Reply:
fwiw if there is no `index` should just not create the index.xml instead of empty contentIndex.json. Because empty contentIndex simply will just break quartz.

Reply:
That would be a bugfix, is it ok to add to this PR? I'll do an early-return if there's no index.md (and add a bypass option, because there are conceivable use cases for Quartz without a homepage) if so.

Reply:
That's fine to do it in here.